### ISSUE TITLE
Using `@latest` should not change the constraint pattern

### ISSUE THREAD

**Issue by Garrett-R** [python-poetry/poetry#3503]
- [X ] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [X ] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.

- **OS version and name**: Ubuntu 18.04
- **Poetry version**: 1.1.4

## Issue

According to [the docs](https://python-poetry.org/docs/cli/#add), to upgrade to the latest version of a package you already have, for example Flask, you can do: `poetry add Flask@latest`.  The issue is that it changes the constraint you have.  So if you start with:

```toml
Flask = "1.1.1"
```

in your `pyproject.toml`, it'll change it to:

```toml
Flask = "^1.1.2"
```

I would expect it to not add the `^`.  This expected behavior matches `npm`.  Doing `npm install typescript@latest` changes `"typescript": "3.8.3"` to `"typescript": "4.1.3"`.

**Comment by sinoroc** [comment]
Yes, the doc could be more explicit about this. On the other hand, the _caret_ notation is basically the default in _poetry_. Keeping version ranges (mostly semver carets) in the `pyproject.toml` and exact pinned versions in the lockfile `poetry.lock` is one of poetry's most prominent features. As far as I know, poetry is not trying to emulate any other tool, not especially trying to be consistent with other tools, but mostly trying to e consistent with itself.

I would agree that maybe there could be a dedicated notation for what you are expecting, maybe `poetry add Library@=latest` or something similar.

**Comment by Garrett-R** [comment]
Even if it's not trying to emulate a specific, I reckon it's still good to match expectations.  And Poetry seems really good at that on the whole (very intuitive, well-designed software).  I bring up `npm` just as evidence pointing to my personal expectation being a generally accepted expectation.

I have no issue with carets being the default in Poetry, but if the user has overridden the default by removing the caret, then adding the caret back is probably not what the user wants.
